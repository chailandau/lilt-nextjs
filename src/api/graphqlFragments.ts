export const INTERNAL_LINK_FRAGMENT = `
internalLink {
    id
    title
    slug
}
`;

export const IMAGE_FRAGMENT = `
id
alt
url
width
height
base64
`;

export const VIDEO_FRAGMENT = `
id
url
placeholderImage {
    ${IMAGE_FRAGMENT}
}
`;

export const MENU_ITEMS_FRAGMENT = `
id
label
linkType
${INTERNAL_LINK_FRAGMENT}
externalLink
`;

export const CTA_FRAGMENT = `
id
label
linkType
${INTERNAL_LINK_FRAGMENT}
externalLink
color
`;

export const HERO_FRAGMENT = `
...on HeroBlock {
    __typename
    id
    hero {
        id
        heading
        subheading
        content
        cta {
            type
            ctaButtons {
                id
                callToAction {
                    ${CTA_FRAGMENT}
                }
            }
            ctaTiles {
                id
                callToActionTile {
                    id
                    linkType
                    label
                    ${INTERNAL_LINK_FRAGMENT}
                    externalLink
                    icon {
                        ${IMAGE_FRAGMENT}
                    }
                }
            }
        }
    }
}
`;
export const ACCORDION_FRAGMENT = `
...on AccordionBlock {
    __typename
    id
    accordion {
        id 
        heading
        accordionItems {
            id
            heading
            content
        }
    }
}
`;

export const CONVERSION_PANEL_FRAGMENT = `
...on ConversionPanelBlock {
    id
    __typename
    conversionPanel {
        id
        heading
        content
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const FEATURE_GRID_FRAGMENT = `
...on FeatureGridBlock {
    id
    __typename
    featureGrid {
        id 
        heading
        featureTiles {
            id
            heading
            image {
                ${IMAGE_FRAGMENT}
            }
            content
        }
    }
}
`;
export const ICON_TILE__GRID_FRAGMENT = `
...on IconTileGridBlock {
    id
    __typename
    iconTileGrid {
        id
        heading
        content
        iconTiles {
            id
            label
            icon {
                ${IMAGE_FRAGMENT}
            }
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const PROCESS_FRAGMENT = `
...on ProcessBlock {
    id
    process {
        id
        steps {
            id
            heading
            content
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const SWITCHBACK_FRAGMENT = `
...on SwitchbackBlock {
    __typename
    id
    switchback {
        id
        image {
            ${IMAGE_FRAGMENT}
        }
        imageSide
        heading
        content
    }
}
`;

const TEXT_GRID = `
 textGrid {
    id
    heading
    content
    textTiles {
        id
        text
    }
    callToAction {
        ${CTA_FRAGMENT}
    }
}
`;

export const TEXT_GRID_FRAGMENT = `
...on TextGridBlock {
    __typename
    id
    ${TEXT_GRID}
}
`;

export const TILE_GRID_FRAGMENT = `
...on TileGridBlock {
    id
    tileGrid {
        id
        heading 
        content
        tiles {
            id
            heading
            content
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;
export const FEATURED_MEDIA_FRAGMENT = `
...on FeaturedMediaBlock {
    id
    __typename
    featuredMedia {
        id
        heading 
        content
        assetType
        image {
            ${IMAGE_FRAGMENT}
        }
        video {
            ${VIDEO_FRAGMENT}
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const SPECIALS_FRAGMENT = `
...on SpecialBlock {
    id
    __typename
    special {
      id
      heading
      subheading
      content
      sideBox
      ${TEXT_GRID}
    }
  }
`;
