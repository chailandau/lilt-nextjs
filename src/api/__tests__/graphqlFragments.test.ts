import * as graphqlFragments from '../graphqlFragments';

describe('Fragments', () => {
    const fragments = {
        ...graphqlFragments
    };

    Object.entries(fragments).forEach(([name, fragment]) => {
        it(`"${name}" matches snapshot`, () => {
            expect(fragment).toMatchSnapshot(name);
        });
    });
});
