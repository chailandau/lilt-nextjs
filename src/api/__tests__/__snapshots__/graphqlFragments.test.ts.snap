// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragments "ACCORDION_FRAGMENT" matches snapshot: ACCORDION_FRAGMENT 1`] = `
"
...on AccordionBlock {
    id
    accordion {
        id 
        heading
        headingTag
        accordionItems {
            id
            heading
            headingTag
            content
        }
    }
}
"
`;

exports[`Fragments "CONVERSION_PANEL_FRAGMENT" matches snapshot: CONVERSION_PANEL_FRAGMENT 1`] = `
"
...on ConversionPanelBlock {
    id
    blockName
    blockType
    conversionPanel {
        id
        heading
        headingTag
        content
        callToAction {
            
id
label
linkType

internalLink {
    id
    title
    slug
}

externalLink
color

        }
    }
}
"
`;

exports[`Fragments "CTA_FRAGMENT" matches snapshot: CTA_FRAGMENT 1`] = `
"
id
label
linkType

internalLink {
    id
    title
    slug
}

externalLink
color
"
`;

exports[`Fragments "FEATURE_GRID_FRAGMENT" matches snapshot: FEATURE_GRID_FRAGMENT 1`] = `
"
...on FeatureGridBlock {
    id
    featureGrid {
        id 
        heading
        headingTag
        featureTiles {
            id
            heading
            headingTag
            image {
                
id
alt
url
width
height

            }
            content
        }
    }
}
"
`;

exports[`Fragments "HERO_FRAGMENT" matches snapshot: HERO_FRAGMENT 1`] = `
"
...on HeroBlock {
    __typename
    id
    hero {
        id
        heading
        headingTag
        subheading
        content
        cta {
            ctaButtons {
                id
                callToAction {
                    
id
label
linkType

internalLink {
    id
    title
    slug
}

externalLink
color

                }
            }
            ctaTiles {
                id
                callToActionTile {
                    id
                    linkType
                    
internalLink {
    id
    title
    slug
}

                    externalLink
                }
            }
        }
    }
}
"
`;

exports[`Fragments "ICON_TILE__GRID_FRAGMENT" matches snapshot: ICON_TILE__GRID_FRAGMENT 1`] = `
"
...on IconTileGridBlock {
    id
    iconTileGrid {
        id
        heading
        headingTag
        content
        iconTiles {
            id
            label
            icon {
                
id
alt
url
width
height

            }
        }
        callToAction {
            
id
label
linkType

internalLink {
    id
    title
    slug
}

externalLink
color

        }
    }
}
"
`;

exports[`Fragments "IMAGE_FRAGMENT" matches snapshot: IMAGE_FRAGMENT 1`] = `
"
id
alt
url
width
height
"
`;

exports[`Fragments "INTERNAL_LINK_FRAGMENT" matches snapshot: INTERNAL_LINK_FRAGMENT 1`] = `
"
internalLink {
    id
    title
    slug
}
"
`;

exports[`Fragments "MENU_ITEMS_FRAGMENT" matches snapshot: MENU_ITEMS_FRAGMENT 1`] = `
"
id
label
linkType

internalLink {
    id
    title
    slug
}

externalLink
"
`;

exports[`Fragments "PROCESS_FRAGMENT" matches snapshot: PROCESS_FRAGMENT 1`] = `
"
...on ProcessBlock {
    id
    process {
        id
        steps {
            id
            heading
            headingTag
            content
        }
        callToAction {
            
id
label
linkType

internalLink {
    id
    title
    slug
}

externalLink
color

        }
    }
}
"
`;

exports[`Fragments "SWITCHBACK_FRAGMENT" matches snapshot: SWITCHBACK_FRAGMENT 1`] = `
"
...on SwitchbackBlock {
    __typename
    id
    switchback {
        id
        image {
            
id
alt
url
width
height

        }
        imageSide
        heading
        headingTag
        content
    }
}
"
`;

exports[`Fragments "TEXT_GRID_FRAGMENT" matches snapshot: TEXT_GRID_FRAGMENT 1`] = `
"
...on TextGridBlock {
    id
    textGrid {
        id
        heading
        headingTag
        content
        textTiles {
            id
            text
        }
        callToAction {
            
id
label
linkType

internalLink {
    id
    title
    slug
}

externalLink
color

        }
    }
}
"
`;

exports[`Fragments "TILE_GRID_FRAGMENT" matches snapshot: TILE_GRID_FRAGMENT 1`] = `
"
...on TileGridBlock {
    id
    tileGrid {
        id
        heading 
        headingTag
        content
        tiles {
            id
            heading
            headingTag
            content
        }
        callToAction {
            
id
label
linkType

internalLink {
    id
    title
    slug
}

externalLink
color

        }
    }
}
"
`;
